File: packages/tui/src/theme/styles.ts
const ansi = (open: number, close: number, value: string): string => {
	if (value.length === 0) {
		return '';
	}

	return `\u001B[${open}m${value}\u001B[${close}m`;
};

export const dim = (value: string): string => ansi(2, 22, value);

export const cyan = (value: string): string => ansi(36, 39, value);

export const green = (value: string): string => ansi(32, 39, value);

export const inverse = (value: string): string => ansi(7, 27, value);

export const black = (value: string): string => ansi(30, 39, value);

export const red = (value: string): string => ansi(31, 39, value);

export const strikethrough = (value: string): string => ansi(9, 29, value);

export const white = (value: string): string => ansi(37, 39, value);

export const yellow = (value: string): string => ansi(33, 39, value);

export const backgroundCyan = (value: string): string => ansi(46, 49, value);

export const backgroundRed = (value: string): string => ansi(41, 49, value);

export const foregroundRgb = (value: string, red: number, green: number, blue: number): string => {
	if (value.length === 0) {
		return '';
	}

	return `\u001B[38;2;${red};${green};${blue}m${value}\u001B[0m`;
};

File: console/shell/internal/theme/styles.go
package theme

import "fmt"

func ansi(open int, close int, value string) string {
	if value == "" {
		return ""
	}

	return fmt.Sprintf("\x1b[%dm%s\x1b[%dm", open, value, close)
}

func Dim(value string) string {
	return ansi(2, 22, value)
}

func Cyan(value string) string {
	return ansi(36, 39, value)
}

func Green(value string) string {
	return ansi(32, 39, value)
}

func Inverse(value string) string {
	return ansi(7, 27, value)
}

func Black(value string) string {
	return ansi(30, 39, value)
}

func Red(value string) string {
	return ansi(31, 39, value)
}

func Strikethrough(value string) string {
	return ansi(9, 29, value)
}

func White(value string) string {
	return ansi(37, 39, value)
}

func Yellow(value string) string {
	return ansi(33, 39, value)
}

func BackgroundCyan(value string) string {
	return ansi(46, 49, value)
}

func BackgroundRed(value string) string {
	return ansi(41, 49, value)
}

func ForegroundRGB(value string, red int, green int, blue int) string {
	if value == "" {
		return ""
	}

	return fmt.Sprintf("\x1b[38;2;%d;%d;%dm%s\x1b[0m", red, green, blue, value)
}
